### ISSUE TITLE
Make the script POSIX-compliant

### ISSUE THREAD

**Issue by Ballasi** [b3nj5m1n/xdg-ninja#11]
I am writing this here as this is being discussed in #6.

In all cases, it'll be up to b3nj5m1n to make the choice, but knowing that the script isn't so big, I think it shouldn't take much effort to make the script POSIX-compliant.

The main reasons why you'd want to have your script POSIX-compliant is that this is standardized and portable across all POSIX (i.e., all Linux, even though not officially POSIX) systems, meaning that shells like bash, or even zsh, dash would work as expected.

Modern desktops most likely uses `bash` so it shouldn't cause a problem using a hard dependency, but as stated before, the amount of effort that has to go into making the script POSIX-compliant isn't so big and could be done easily too. This would ensure everyone can run the script correctly.

**Comment by b3nj5m1n** [comment]
Okay, I've started to work on this, see

- 5eeeda0ad50d4814881e3dfc3930d7905d287794
- 3184a4a4678e5ed471e2678718d5894e43184e88
- 6a67c7120d414a8a893806d4f19294e3c8354902
- 22566a147329dacfb1e364f0f850616c8c22dc15
- f6fa95e86ee65419ec3cceb473af5e62df405038
- 4efb805b58bc6c9fd2d008b75e8990622fe4eb9c
- bf9f62a814c9860d53bfb71d6e5b5d23a82c6922

---

The script should b eclose to POSIX compliant now, according to shellcheck there are two things remaining that are bash-specific, and I'm not sure how to solve them, so maybe you (or someone else) can help me.

---

The first issue is [here](https://github.com/b3nj5m1n/xdg-ninja/blob/9484de3ee968e2a01b8691adc95a012f2c1cf787/xdg-ninja.sh#L175):
```bash
    while IFS= read -r file; do
        check_file "$file" "$NAME"
    done <<<"$(printf "%s" "$INPUT" | jq -rc '.files[]')"
```
Apparently, the `<<<` syntax isn't supported in POSIX.

---

The second one, rather unsurprisingly, is [here](https://github.com/b3nj5m1n/xdg-ninja/blob/9484de3ee968e2a01b8691adc95a012f2c1cf787/xdg-ninja.sh#L182):
```bash
	for prog_filename in "$(dirname "${BASH_SOURCE[0]}")"/programs/*; do
```
I mean, it has bash in the name. I think I've found a solution to this one already, I'll look into it soon.

**Comment by b3nj5m1n** [comment]
Okay, those last two things are implemented with b6d3bc4e2f25365294a4bcd2ecebee64b919770c, thanks to the help from @2641a40fd44383320adde4b027a1d0b03bd550.

I've checked with shellcheck, and it looks like xdg-ninja is now fully POSIX compliant! 🎉

Would be great if you could double-check and then close this issue.

**Comment by Ballasi** [comment]
Everything looks fine for me :tada: 
I am closing the issue!
